packages/cli/test/e2e/cypress/integration/build.ts

```diff
@@ -0,0 +1,65 @@
+const BLITZ_PATH = "../blitz/bin/blitz"
+const TEST_APP_NAME = "app"
+
+// waiting 2 minutes for the command to execute or test breaks
+const execConfig = {
+  log: true,
+  timeout: 120000,
+}
+
+const buildAppCommand = (
+  appName: string = TEST_APP_NAME,
+  packageManager: "yarn" | "npm",
+  template: "full" | "minimal",
+  formLib: "react-final-form" | "react-hook-form" | "formik",
+  language: "typescript" | "javascript",
+): void => {
+  cy.exec(
+    `${BLITZ_PATH} new ${appName} --${packageManager} --template=${template}  --form=${formLib} --language=${language}`,
+    execConfig,
+  )
+  cy.writeFile(
+    `./${TEST_APP_NAME}/types.ts`,
+    "declare module 'react' { interface StyleHTMLAttributes<T> extends React.HTMLAttributes<T> { jsx?: boolean; global?: boolean }}",
+    {encoding: "utf8", flag: "a+"},
+  )
+
+  cy.readFile("../blitz/package.json").then((packageJson) => {
+    cy.exec(
+      `cd ${appName} && ${packageManager} add blitz@${packageJson.dependencies["@blitzjs/cli"]} --save`,
+    )
+    cy.readFile(`./${appName}/package.json`).then((testAppPackageJson) => {
+      expect(testAppPackageJson.dependencies["blitz"]).eqls(
+        packageJson.dependencies["@blitzjs/cli"],
+      )
+    })
```

Comment:
I have some doubts about whether this is doing what we want it too. Ultimately what we want is that none of the `blitz`, `@blitzjs/*`, or `next` packages should be inside the `<test-app-dir>/node_modules/` folder. All of these packages are already linked in `<monorepo-root>/node_modules`. So if those packages are not in the test app node_modules, they will automatically be resolved to the monorepo root.

We need to carefully confirm this. I suspect what's happening with the current code is that it's installing the given version from NPM, which is not what we want.

Reply:
It's installing the version from npm indeed. my attention behind this was to follow according to steps that you mentioned, I probably miss understood what you meant by : 
- get blitz version from packages/blitz/package.json version field
- set blitz version in the new test app package.json to match the version from last step. ( i'm installing the corresponding blitz version in this step )

Reply:
Ok yeah. Maybe another approach is to use the `file:...` syntax and point it to the blitz package?

Reply:
> Ok yeah. Maybe another approach is to use the `file:...` syntax and point it to the blitz package?

Not sure I get totally what you mean, as I understood this was made because we want blitz version in the repo to be the same as the generated app that we are testing, since the app is generated using `packages/blitz`, should I assume that the generated app is with same version as `packages/blitz` ? 

And what do you mean exactly by `file:...`

Reply:
I think what Brandon meant is local paths (more here: https://docs.npmjs.com/cli/v7/configuring-npm/package-json#local-paths), so having something like:
```
{
  "name": "tests",
  "dependencies": {
    "blitz": "file:../../packages/blitz"
  }
}
```
You can run `npm i <path to package>` and it will add it in the above way.